pkg/controller/siddhiprocess/artifact/artifacts.go

```diff
@@ -60,31 +67,32 @@ func (rsp *ReconcileSiddhiProcess) CreateOrUpdateCM(
 			Kind:       "ConfigMap",
 		},
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      configMapName,
-			Namespace: sp.Namespace,
+			Name:      name,
+			Namespace: namespace,
 		},
 		Data: data,
 	}
-	controllerutil.SetControllerReference(sp, configMap, rsp.scheme)
-	_, err = controllerutil.CreateOrUpdate(context.TODO(), rsp.client, configMap, ConfigMapMutateFunc(data))
+	controllerutil.SetControllerReference(owner, configMap, k.Scheme)
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), k.Client, configMap, ConfigMapMutateFunc(data))
 	return err
 }
 
-// CreateOrUpdateIngress creates a NGINX Ingress load balancer object called siddhi
-func (rsp *ReconcileSiddhiProcess) CreateOrUpdateIngress(
-	sp *siddhiv1alpha2.SiddhiProcess,
-	siddhiApp SiddhiApp,
-	configs Configs,
+// CreateOrUpdateIngress creates an NGINX Ingress load balancer object called siddhi
+func (k *KubeClient) CreateOrUpdateIngress(
+	namespace string,
+	serviceName string,
+	tls string,
+	containerPorts []corev1.ContainerPort,
 ) (operationResult controllerutil.OperationResult, err error) {
 
 	var ingressPaths []extensionsv1beta1.HTTPIngressPath
-	for _, port := range siddhiApp.ContainerPorts {
-		path := "/" + strings.ToLower(siddhiApp.Name) +
+	for _, port := range containerPorts {
+		path := "/" + strings.ToLower(serviceName) +
```

Comment:
You can use `path` package to build the URLs. https://golang.org/pkg/path/

Reply:
Fixed in https://github.com/siddhi-io/siddhi-operator/pull/123/commits/785753fcea0dd5dcef62c63fe739e8e53b54e03e